python/jupytercad_lab/jupytercad_lab/notebook/cad_document.py

```diff
@@ -166,31 +165,55 @@ def remove_annotation(self, annotation_id: str) -> None:
             with self.ydoc.transaction() as t:
                 self._metadata.pop(t, annotation_id, None)
 
+    def get_object_by_name(self, object_name: str) -> Optional[Dict]:
+        """
+        Retrieve an object by its name.
+
+        :param object_name: Name of the object.
+        :return: The object if it exists, otherwise None.
+        """
+        for obj in self._options.get("objects", []):
+            if obj["name"] == object_name:
+                return obj
+        return None
+
+    def update_object_by_name(
+        self, transaction, object_name: str, updated_obj: Dict
+    ) -> None:
+        """
+        Update an object by its name within a transaction.
+
+        :param transaction: The YDoc transaction.
+        :param object_name: Name of the object to update.
+        :param updated_obj: The updated object data.
+        """
+        objects = self._options.get("objects", [])
+
+        for i, obj in enumerate(objects):
+            if obj["name"] == object_name:
+                objects[i] = updated_obj
+                break
+
+        self._options.set(transaction, "objects", objects)
```

Comment:
Did you try this method? I'm not sure it will work since `pycrdt.Map` does not have the `set` method. Let's remove the Python bits and do it in a follow-up PR.

Reply:
> Let's remove the Python bits and do it in a follow-up PR.

Sure, that sounds good. On it

Reply:
Done in [`874b49f`](https://github.com/jupytercad/JupyterCAD/pull/413/commits/874b49f90c2cc4a0b06ebfbd9c16ec6dde7f8e50)